state.Pub is nil

After latest merge in wax branch starting factomd result in this error:
```
FNode0 Loading from /home/paul/.factom/m2/FastBoot_LOCAL_v13.db
              FNode0 Load state from /home/paul/.factom/m2/FastBoot_LOCAL_v13.db
State move between non-sequential heights from 0 to 71001
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb28ee2]

goroutine 95 [running]:
github.com/FactomProject/factomd/state.(*State).MoveStateToHeight(0xc000164000, 0x11559, 0x0)
	github.com/FactomProject/factomd/state/stateConsensus.go:895 +0x712
github.com/FactomProject/factomd/state.(*SaveState).RestoreFactomdState(0xc000212820, 0xc000164000)
	github.com/FactomProject/factomd/state/saveAndRestore.go:669 +0xe72
github.com/FactomProject/factomd/state.(*StateSaverStruct).LoadDBStateList(0xc000164dd8, 0xc000164000, 0xc000120c80, 0x7ffef761f0fd, 0x5, 0x0, 0x0)
	github.com/FactomProject/factomd/state/stateSaver.go:112 +0x3e6
github.com/FactomProject/factomd/state.(*State).Initialize(0xc000164000, 0x10db360, 0xc000304000, 0x10d9620, 0x17adac8)
	github.com/FactomProject/factomd/state/factory.go:654 +0x1689
github.com/FactomProject/factomd/engine.makeServer(0xc000304000, 0x176ee80, 0xc0001601e0)
	github.com/FactomProject/factomd/engine/NetStart.go:378 +0xe0
github.com/FactomProject/factomd/engine.initEngine.func1(0xc000304000)
	github.com/FactomProject/factomd/engine/NetStart.go:174 +0x34
github.com/FactomProject/factomd/engine.NetStart(0xc000304000, 0x176ee80, 0x80001)
	github.com/FactomProject/factomd/engine/NetStart.go:154 +0x57
github.com/FactomProject/factomd/engine.Factomd(0xc000304000, 0x176ee80, 0xc00010ed01)
	github.com/FactomProject/factomd/engine/factomd.go:53 +0x2eb
github.com/FactomProject/factomd/engine.Run.func1(0xc000304000)
	github.com/FactomProject/factomd/engine/factomd.go:36 +0x43
github.com/FactomProject/factomd/modules/registry.(*process).bindCallbacks.func1(0xc0001ef160, 0xc000304000, 0xc0001f5030)
	github.com/FactomProject/factomd/modules/registry/registry.go:70 +0x34
created by github.com/FactomProject/factomd/modules/registry.(*process).bindCallbacks
	github.com/FactomProject/factomd/modules/registry/registry.go:68 +0x53
```
Apparently s.Pub here `github.com/FactomProject/factomd/state/stateConsensus.go:895` is nil

--- a/engine/NetStart.go
+++ b/engine/NetStart.go
@@ -375,9 +375,9 @@ func makeServer(w *worker.Thread, p *globals.FactomParams) (node *fnode.FactomNo
 
 	// Election factory was created and passed int to avoid import loop
 	//node.State.BindPublishers() // REVIEW: has this been fully refactored?
-	node.State.Initialize(w, new(electionMsgs.ElectionsFactory))
 	node.State.NameInit(node, node.State.GetFactomNodeName()+"STATE", reflect.TypeOf(node.State).String())
 	node.State.BuildPubRegistry()
+	node.State.Initialize(w, new(electionMsgs.ElectionsFactory))
 
 	state0Init.Do(func() {
 		logPort = p.LogPort

--- a/state/stateConsensus.go
+++ b/state/stateConsensus.go
@@ -881,9 +881,7 @@ func (s *State) MoveStateToHeight(dbheight uint32, newMinute int) {
 					DirectoryBlockHeader: dbstate.DirectoryBlock.GetHeader(),
 					Timestamp:            s.GetTimestamp(),
 				}
-				if s.Pub != nil {
-					s.Pub.Directory.Write(&directory)
-				}
+				s.Pub.Directory.Write(&directory)
 				s.LogPrintf("leader", "%v", directory)
 
 			}
@@ -949,9 +947,7 @@ func (s *State) MoveStateToHeight(dbheight uint32, newMinute int) {
 	s.DBSigLimit = s.EOMLimit               // We add or remove server only on block boundaries
 	s.LogPrintf("dbstateprocess", "MoveStateToHeight(%d-:-%d) leader=%v leaderPL=%p, leaderVMIndex=%d", dbheight, newMinute, s.Leader, s.LeaderPL, s.LeaderVMIndex)
 
-	if s.Pub != nil { // FIXME: should coordinate threads to start processing only after publish
-		s.Pub.BlkSeq.Write(&events.DBHT{DBHeight: s.LLeaderHeight, Minute: s.CurrentMinute})
-	}
+	s.Pub.BlkSeq.Write(&events.DBHT{DBHeight: s.LLeaderHeight, Minute: s.CurrentMinute})
 
 	s.Hold.ExecuteForNewHeight(s.LLeaderHeight, s.CurrentMinute) // execute held inMessages
 	s.Hold.Review()                                              // cleanup old inMessages

